src/workerd/api/queue.c++

```diff
@@ -523,6 +523,12 @@ kj::Promise<WorkerInterface::CustomEvent::Result> QueueCustomEventImpl::run(
   // reuse the scheduled worker logic and timeout for now.
   auto completed = co_await incomingRequest->finishScheduled();
 
+  // Log some debug info if the queue handler completed but other waituntils did not
+  if (!completed && queueEventHolder->event->getIsQueueHandlerComplete()) {
+    auto taskTrace = incomingRequest->getContext().getWaitUntilTasks().trace();
+    KJ_LOG(WARNING, kj::str("Queue event timed out with lingering waitUntil tasks:\n", taskTrace));
```

Comment:
Do the newlines render alright here? Was there specific thinking behind pre-rendering it into a string vs letting kj do its thing with code like `KJ_LOG(WARNING, "Queue event timed out with lingering waitUntil tasks", taskTrace));`

Did you manage to try this out (or you could use any help trying it out) locally?